### PR TITLE
fix!: query validation logic

### PR DESCRIPTION
**Pull request by QuantumExplorer**
## Issue being fixed or feature implemented
Queries used to be validated both in rs-drive and js-drive. We decided to only do it in rs-drive.

## What was done?
In order to do that we needed to port some query validation and tests from js-drive.

## Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: drive/src/query/mod.rs [hunk 1]
```diff
@@ -946,4 +1002,369 @@ mod tests {
...
+
+        let where_cbor = common::value_to_cbor(query_value, None);
+        let query = DriveQuery::from_cbor(where_cbor.as_slice(), &contract, &document_type)
+            .expect("fields of queries length must be under 256 bytes long");
+        query
+            .execute_no_proof(&drive, None)
+            .expect_err("fields of queries length must be under 256 bytes long");
+    }
+
+    // Eventually we want to error with weird Null values
```
Comment (shumkov) [inline drive/src/query/mod.rs]: Add "TODO"

File: drive/src/query/mod.rs [hunk 2]
```diff
@@ -946,4 +1002,369 @@ mod tests {
...
+    //
+    //     let where_cbor = common::value_to_cbor(query_value, None);
+    //     let query = DriveQuery::from_cbor(where_cbor.as_slice(), &contract, &document_type)
+    //         .expect("The query itself should be valid for a null type");
+    //     query
+    //         .execute_no_proof(&drive, None)
+    //         .expect_err("a Null value doesn't make sense for an integer");
+    // }
+
+    // Eventually we want to error with weird Null values
```
Comment (shumkov) [inline drive/src/query/mod.rs]: Add "TODO"

File: drive/src/query/mod.rs (hunk 1 above)
Comment (QuantumExplorer) [reply drive/src/query/mod.rs]: There's no point, as we write all queries at some point in Rust. There's a massive todo here.

**Review by shumkov — APPROVED** [review]
Looks good to me! 👍

**Pull request merged**